CONTRIBUTING.md

```diff
@@ -1,6 +1,132 @@
+
+1. Regularly fetch latest stingray development version `stingray/master` from GitHub.
+2. Make a new feature branch. **Recommended:** Use virtual environments to work on branch.
+3. Editing Workflow:
+    1. One commit per logical changes.
+    2. Run tests to make sure that changes don't break existing code.
+    3. Code should have appropriate docstring.
+    4. Update appropriate documentation if necessary and test it on sphinx.
+    5. Write a test for code changes.
+    6. If  modifications require more than one commit, break changes into smaller commits.
+    7. Push the code on your remote(forked) repository.
+4. All code changes should be submitted via PR's(i.e. fork, branch, work on stuff, just submit pull request).
+Code Reviews are super-useful: another contributor can review the code, which means both the contributor and reviewer will be up to date with how everything fits together, and can get better by reading each other's code! :)
+5. Take feedback and make changes/revise the PR as asked.
+6. Don't merge using web interface if your branch falls behind master. Fetch and rebase.
+
+## Coding Guidelines
+
+---
+
+### Compatibility and Dependencies
+
+- **Compatibility:** All code must be compatible with **Python 3.5** **or later**. Usage of six, __future__, and 2to3 is not longer acceptable.
+- **Dependency Management:**
+    - The core package and affiliated packages should be importable with no dependencies other than the [Python Standard Library](https://docs.python.org/3/library/index.html), [astropy](https://docs.astropy.org/en/stable/), [numpy](https://numpy.org/doc/stable/)>=1.11.0, [scipy](https://www.scipy.org/docs.html)>=0.18.0, [matplotlib](https://matplotlib.org/contents.html)>=1.3.
+    - Additional dependencies - such as SciPy, Matplotlib, or other third-party packages - are allowed for sub-modules or in function calls, but they must be noted in the package documentation and should only affect the relevant component. In functions and methods, the optional dependency should use a normal `import` statement, which will raise an `ImportError` if the dependency is not available.
+
+### Coding Style and Conventions
+
+- **Style Guide:**
+    - Follow the [PEP8 style guide](https://www.python.org/dev/peps/pep-0008/). Follow the existing coding style within the sub-package and avoid changes that are purely stylistic.
+    - Indentation should be **ONLY** with **four spaces** no mixing of tabs-and-spaces.
+    - Maximum line length should be **79** characters unless doing so makes the code unreadable, ugly.
+    - Functions and methods should be lower-case only, and separated by a '_'  in case of multiple words eg. `my_new_method`.
+    - Use verbose variable names(readability > economy). Only loop iteration variables are allowed to be a single letter.
+    - Classes start with an upper-case letter and use CamelCase eg. `MyNewClass`.
+    - Line comments should start with a single #.
```

Comment:
Inline comments should start with two spaces and a "#"
```
var  = 'blahblah'  # This is an inline comment
```

Reply:
Changed